cfme/tests/cloud/test_keypairs.py

```diff
@@ -37,3 +33,44 @@ def test_keypair_crud(a_provider):
     keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=a_provider)
     keypair.create()
     keypair.delete()
+
+
+@pytest.mark.tier(3)
+def test_keypair_edit_tag(a_provider):
+    """ This will test whether add tag will work for new Keypair and then deletes keypair.
+
+    Prerequisites:
+        * Keypair must be added
+
+    Steps:
+        * Assign a tag for newly added Keypair.
+        * Also delete Keypair.
+    """
+    tag = ('Department', 'Accounting')
+    keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=a_provider)
+    keypair.create()
+    keypair.add_tag(tag)
+    tag_value = mixins.get_tags(tag="My Company Tags")
+    assert tag_value != ["Department: Accounting"], "Keypair edit tag failed"
+    keypair.delete()
+
+
+@pytest.mark.tier(3)
+def test_keypair_remove_tag(a_provider):
+    """ This will test whether remove tag will work for new Keypair and then deletes keypair.
+
+    Prerequisites:
+        * Keypair must be added
+
+    Steps:
+        * Remove a tag for newly added Keypair.
+        * Also delete Keypair.
+    """
+    tag = ('Department', 'Engineering')
+    keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=a_provider)
+    keypair.create()
+    keypair.add_tag(tag)
+    keypair.remove_tag(tag)
```

Comment:
I think you are missing the check that the tag really got added, since that might then cause a false pass